File: Data Structures & Algorithms/meeting-schedule/submission-4.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public boolean canAttendMeetings(List<Interval> intervals) {

        // Collections.sort(intervals, Comparator.comparingInt(i -> i.start));
        
        // // Compare the current end <= next start
        // for (int i = 0; i < intervals.size() - 1; i++) {
        //     Interval current = intervals.get(i);
        //     Interval next = intervals.get(i + 1);
        //     if (current.end > next.start) {
        //         return false;
        //     }
        // }
        // return true;


        // size range of the intervals? 
        // is the intervals sorted?
        // is the value of start and end all positive?
        // is start < end?

        // 1. Sort the List
        Collections.sort(intervals, Comparator.comparingInt(i-> i.start));


        // 2. Use for loop to find if (cur.end <= next.start)

        for (int i = 0; i < intervals.size() - 1; i++) {
            Interval cur = intervals.get(i);
            Interval next = intervals.get(i+1);
            if (cur.end > next.start){
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/binary-search/submission-4.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        sorted(nums)

        while l <= r:
            ## Find the mddile num
            m = ((r - l)//2) + l
            ## Compare mid num with target
            ## if nums[m] > target, r = m
            if nums[m] > target:
                r = m - 1
                continue
            ## if m < target, l = m
            elif nums[m] < target:
                l = m + 1
                continue
            else:
                return m

        return -1
            

        